core/util/eventtarget.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * noVNC: HTML5 VNC client
+ * Copyright 2017 Pierre Ossman for Cendio AB
+ * Licensed under MPL 2.0 (see LICENSE.txt)
+ *
+ * See README.md for usage and integration instructions.
+ */
+
+export var EventTargetMixin = {
+    _listeners: null,
+
+   addEventListener: function(type, callback) {
+      if (!this._listeners) {
+         this._listeners = {};
+      }
+      if (!(type in this._listeners)) {
+         this._listeners[type] = [];
+      }
+      this._listeners[type].push(callback);
+   },
+
+   removeEventListener: function(type, callback) {
+      if (!this._listeners || !(type in this._listeners)) {
+         return;
+      }
+      var stack = this._listeners[type];
+      for (var i = 0, l = stack.length; i < l; i++) {
+         if (stack[i] === callback) {
```

Comment:
make this a `Set` to allow easy removal.

Reply:
Alright, I'll give it a try.